### PR TITLE
CBL-3778: Collection SG test, "Auto Purge Enabled - Filter Removed Re…

### PR DESCRIPTION
**Pull request by jianminzhao**
…vision"

### CODE REVIEW

File: Replicator/tests/ReplicatorCollectionSGTest.cc [hunk 1]
```diff
@@ -1070,3 +1099,121 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Auto Purge Enabled - Remove Doc Fr
...
+         int docsEndedPurge = 0;
+         int pullFilterTotal = 0;
+         int pullFilterPurge = 0;
+         void reset() {
+             docsEndedTotal = 0;
+             docsEndedPurge = 0;
+             pullFilterTotal = 0;
+             pullFilterPurge = 0;
+         }
+     } cbContext;
```
Comment (borrrden) [inline Replicator/tests/ReplicatorCollectionSGTest.cc:1139]: This is tons of lines duplicated with other tests, which is a recipe for unreliable maintainability.

**Review by borrrden — APPROVED** [review]
Careful of this building tech debt in the form of copy pasted code

File: Replicator/tests/ReplicatorCollectionSGTest.cc (hunk 1 above)
Comment (jianminzhao) [reply Replicator/tests/ReplicatorCollectionSGTest.cc:1139]: Do you refer to the above or below?

Comment (borrrden) [reply Replicator/tests/ReplicatorCollectionSGTest.cc:1139]: Always above....below does not show up in the snippet above

Comment (jianminzhao) [reply Replicator/tests/ReplicatorCollectionSGTest.cc:1139]: It's just like 4 local variables. Thought to take them out but could not find a meaningful standalone description and there are only 3 or 4 tests may share.

**Pull request merged**